Automattic/wp-calypso
# ExternalLink to packages (Step 2): Temporarily shim old Calypso ExternalLink component

Related to https://github.com/Automattic/wp-calypso/issues/88181
Branched from https://github.com/Automattic/wp-calypso/pull/88236

## Proposed Changes

- We point the old `ExternalLink` component to the new one in `@automattic/components`, allowing us to transition from the old code to the new in smaller pieces and without risking updates to the old component in the interim.
- Remove old docs and files

## Testing Instructions



- Ensure `/devdocs` renders the `ExternalLink` fine
- Test instances of the `ExternalLink` and `ExternalLinkWithTracking` components in Calypso. There are about half a million instances, so let's just test a few. 
    - For the `ExternalLinkWithTracking` cases, ensure the Tracks event is being sent properly (it now uses the utility from `@automattic/calypso-analytics`, not the locally defined one as done previously)

Examples of both:

- **`ExternalLinkWithTracking` - in `/setup/newsletter/plans` (with Plans FAQ enabled)**

<img width="450" alt="Screenshot 2024-03-12 at 11 41 09 AM" src="https://github.com/Automattic/wp-calypso/assets/1705499/8dce20ee-af31-41e0-b59c-58c03dae5bb5">

- **`ExternalLink` - in `/people/team/[ site ]`**

<img width="450" alt="Screenshot 2024-03-12 at 11 51 23 AM" src="https://github.com/Automattic/wp-calypso/assets/1705499/84f3533f-c7c9-436e-b0fa-eab360949469">


## Pre-merge Checklist
